packages/now-node/src/dev-server.ts

```diff
@@ -8,17 +8,31 @@ if (!entrypoint) {
   throw new Error('`VERCEL_DEV_ENTRYPOINT` must be defined');
 }
 
+import { join } from 'path';
 import { register } from 'ts-node';
 
-// Use the project's version of TypeScript if available,
-// otherwise fall back to using the copy that `@vercel/node` uses.
-let compiler: string;
-try {
-  compiler = require.resolve('typescript', {
-    paths: [process.cwd()],
+const resolveTypescript = (p: string): string =>
+  require.resolve('typescript', {
+    paths: [p],
   });
-} catch (e) {
-  compiler = 'typescript';
+
+// Use the project's version of TypeScript if available
+let compiler = '';
+try {
+  const resolved = resolveTypescript(process.cwd());
+
+  // Only use the user's resolved "typescript" if it's
+  // located *within* the project root directory
```

Comment:
What does "project root directory" mean here?
We have `workPath` and `repoRoot` in the `build()` function for monorepos so does this represent `workPath`?

Reply:
Ya, `workPath`.